src/core/public/chrome/nav_links/to_nav_link.ts

```diff
@@ -29,6 +34,28 @@ export function toNavLink(app: PublicAppInfo, basePath: IBasePath): NavLinkWrapp
   });
 }
 
+export function toNavDeepLink(
+  app: PublicAppInfo,
+  deepLink: PublicAppDeepLinkInfo,
+  basePath: IBasePath
+): NavLinkWrapper {
+  const defaultNavStatus = deepLink.navLinkStatus === AppNavLinkStatus.default;
+  const relativeBaseUrl = basePath.prepend(`${app.appRoute!}${deepLink.path!}`);
```

Comment:
May want to use `appendAppPath` instead of plain string concatenation here (`src/core/public/application/utils/append_app_path.ts`)